satpy/readers/satpy_cf_nc.py

```diff
@@ -311,11 +311,16 @@ def get_dataset(self, ds_id, ds_info):
         if name != ds_id["name"]:
             data = data.rename(ds_id["name"])
         data.attrs.update(nc.attrs)  # For now add global attributes to all datasets
-        if "orbital_parameters" in data.attrs:
-            data.attrs["orbital_parameters"] = _str2dict(data.attrs["orbital_parameters"])
-
+        self._decode_dict_type_attrs(data)
         return data
 
+    def _decode_dict_type_attrs(self, data):
+        for key in ["orbital_parameters", "time_parameters"]:
```

Comment:
```suggestion
        for key in ("orbital_parameters", "time_parameters"):
```

Reply:
Updated the method to convert any string starting with `{` to dict (see https://github.com/pytroll/satpy/issues/2609#issuecomment-1778937364)